code/go/0chain.net/smartcontract/dbs/event/delegate_pool.go

```
@@ -63,14 +65,35 @@ func (sp *DelegatePool) exists(edb *EventDb) (bool, error) {
 	return true, nil
 }
 
-func (edb *EventDb) updateReward(reward int64, dp DelegatePool) error {
+func (edb *EventDb) updateReward(reward int64, dp DelegatePool) (err error) {
 	dpu := dbs.NewDelegatePoolUpdate(dp.PoolID, dp.ProviderID, dp.ProviderType)
 
-	if dp.ProviderType == int(spenum.Blobber) && reward < 0 {
-		dpu.Updates["total_penalty"] = dp.TotalPenalty - reward
+	if reward < 0 {
+		reward = -reward
+		if dp.ProviderType == int(spenum.Blobber) {
+			dpu.Updates["total_penalty"], err = currency.MinusInt64(dp.TotalPenalty, reward)
+			if err != nil {
+				return err
+			}
+		} else {
+			dpu.Updates["reward"], err = currency.MinusInt64(dp.Reward, reward)
+			if err != nil {
+				return err
+			}
+			dpu.Updates["total_reward"], err = currency.MinusInt64(dp.TotalReward, reward)
+			if err != nil {
+				return err
+			}
+		}
 	} else {
-		dpu.Updates["reward"] = dp.Reward + reward
-		dpu.Updates["total_reward"] = dp.TotalReward + reward
+		dpu.Updates["reward"], err = currency.AddInt64(dp.Reward, reward)
+		if err != nil {
+			return err
+		}
+		dpu.Updates["total_reward"], err = currency.AddInt64(dp.TotalReward, reward)
+		if err != nil {
+			return err
+		}
 	}
 	if err := edb.updateDelegatePool(*dpu); err != nil {
 		return nil
```

seems error is not returned